Taarafo.Core.Tests.Unit/Services/Foundations/Posts/PostServiceTests.Logic.RetrieveById.cs

```diff
@@ -0,0 +1,51 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Force.DeepCloner;
+using Moq;
+using Taarafo.Core.Models.Posts;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Posts
+{
+    public partial class PostServiceTests
+    {
+        [Fact]
+        public async Task ShouldRetrievePostByIdAsync()
+        {
+            // given
+            Post randomPosts = CreateRandomPost();
+            Guid inputPostId = Guid.NewGuid();
```

Comment:
Since this is the happy path. You will need two variables. `randomPostId` and then `inputPostId`